### PR TITLE
Upgrade Rubygems according to bundler-audit

### PR DESCRIPTION
**Pull request by JuanitoFatas**
This Pull Request fix vulnerable gems according to [bundler-audit](https://github.com/rubysec/bundler-audit):

```
$ bundle-audit
Insecure Source URI found: git://github.com/jnicklas/capybara.git
Insecure Source URI found: git://github.com/matthewrudy/oauth2.git
Insecure Source URI found: git://github.com/plataformatec/devise.git
Insecure Source URI found: git://github.com/rails/rails.git
Insecure Source URI found: git://github.com/rails/record_tag_helper.git
Insecure Source URI found: git://github.com/rails/web-console.git
Name: devise
Version: 3.5.3
Advisory: CVE-2015-8314
Criticality: Unknown
URL: http://blog.plataformatec.com.br/2016/01/improve-remember-me-cookie-expiration-in-devise/
Title: Devise Gem for Ruby Unauthorized Access Using Remember Me Cookie
Solution: upgrade to >= 3.5.4

Vulnerabilities found!
```

- Fix Insecure Source URI

  Use https instead of `github` option, use an insecure source could be subjected to MITM attack.

- Use released web-console 3.1.1

- **Security Updates:**

    - Fixed: [devise](https://github.com/plataformatec/devise), [3.5.3...4.0.0.pre.dev](https://github.com/plataformatec/devise/compare/a321282ccc93d73f9d0f7a1319d3fcab27d25b20...b97b3e6e3b570324467ac84d807111837943ec20) ([CHANGELOG](https://github.com/plataformatec/devise/blob/master/CHANGELOG.md))
      * http://rubysec.com/advisories/CVE-2015-8314
    - Fixed: [nokogiri](https://github.com/sparklemotion/nokogiri), [1.6.7.1...1.6.7.2](https://github.com/sparklemotion/nokogiri/compare/v1.6.7.1...v1.6.7.2) ([CHANGELOG](https://github.com/sparklemotion/nokogiri/blob/master/CHANGELOG.ja.rdoc))
      * http://rubysec.com/advisories/CVE-2015-7499

--

By the way,
I ran this repo with a free service called https://www.deppbot.com to find out about this, see the [Pull Request issued by deppbot](https://github.com/JuanitoFatas/codetriage/pull/1). :blush:

### CODE REVIEW

**Comment by schneems** [comment]
Isn't this a bug in bundled that it doesn't use the https for "github" dependencies?

**Comment by JuanitoFatas** [comment]
> Isn't this a bug in bundled that it doesn't use the https for "github" dependencies?

It is fixed in Bundler v2.0. But for compatibility Bundler v1.0 does not have this. See this [Pull Request](https://github.com/bundler/bundler/pull/4127/files).

**Comment by schneems** [comment]
Looks like I introduced a merge conflict, can you rebase. Sorry! :speak_no_evil:

**Comment by JuanitoFatas** [comment]
Regarding to test failutres: seems `ActionDispatch::IntegrationTest` cannot find routing helpers.

```
Error:
UserUnsubscribeTest#test_unsubscribing_more_than_once:
NoMethodError: undefined method `token_delete_user_path' for #<UserUnsubscribeTest:0x007fed113a0e70>
    test/integration/user_unsubscribes_test.rb:25:in `block in <class:UserUnsubscribeTest>'
```

File: Gemfile.lock [hunk 1]
```diff
@@ -21,10 +22,10 @@ GIT
       xpath (~> 2.0)
 
 GIT
-  remote: git://github.com/plataformatec/devise.git
-  revision: a321282ccc93d73f9d0f7a1319d3fcab27d25b20
+  remote: https://github.com/plataformatec/devise.git
+  revision: ffe9d6d406e79108cf32a2c6a1d0b3828849c40b
   specs:
-    devise (3.5.3)
+    devise (4.0.0.pre.dev)
```
Comment (prathamesh-sonpatki) [inline Gemfile.lock]: @JuanitoFatas This might be issue of test failures? Can we use released gem instead of this?

File: Gemfile.lock [hunk 2]
```diff
@@ -32,13 +33,14 @@ GIT
       warden (~> 1.2.3)
 
 GIT
-  remote: git://github.com/rails/rails.git
-  revision: c48790e4f6d3b733c55e7812f10ccad220ee7f81
+  remote: https://github.com/rails/rails.git
+  revision: 4de092240004e31c512810b19e1b7a83e1ddc77e
```
Comment (prathamesh-sonpatki) [inline Gemfile.lock]: Are we updating Rails too? Is it needed?

File: Gemfile.lock [hunk 3]
```diff
@@ -10,10 +10,11 @@ GIT
       rack (>= 1.2, < 3)
 
 GIT
-  remote: git://github.com/jnicklas/capybara.git
-  revision: 4b3093f4bd12a682b41d8e88d3958d33e70f4662
+  remote: https://github.com/jnicklas/capybara.git
+  revision: 3f5c36f1a45e0f05365539524ee17adb20a7c8d5
   specs:
```
Comment (prathamesh-sonpatki) [inline Gemfile.lock]: Is there any need to update capybara?

Comment (JuanitoFatas) [inline Gemfile.lock]: If you change master Gemfile, capybara gem line from `github: ...` to `git: ...`, bundler will refetch the repo, hence this change.

File: Gemfile.lock (hunk 2 above)
Comment (JuanitoFatas) [inline Gemfile.lock]: Same as https://github.com/codetriage/codetriage/pull/439#discussion_r51238019

File: Gemfile.lock (hunk 1 above)
Comment (JuanitoFatas) [inline Gemfile.lock]: > [Only Devise master supports Rails 5](https://github.com/plataformatec/devise/issues/3879#issuecomment-168376663).

**Comment by JuanitoFatas** [comment]
I opened an issue on devise regarding to this: https://github.com/plataformatec/devise/issues/3922.

**Comment by prathamesh-sonpatki** [comment]
@JuanitoFatas I got it now :smile:  Without running `bundle update` we can't get the updated soruce in Gemfile.lock right? But that updates too many gems in the process. So can you manually update Gemfile.lock to not update any git sha but only update git source :grin:. Looks good to :ship: after that

**Comment by winston** [comment]
> So can you manually update Gemfile.lock to not update any git sha but only update git source

@prathamesh-sonpatki Is the intent to lock those gems at the specific SHA?

**Comment by prathamesh-sonpatki** [comment]
@winston At-least for those gems which are causing failures after updating SHA. We are on 5.0 beta so things are unstable anyways. I think almost all of the gems will point to released versions once they are released. Right now we have too many entries pointing to master branches and if we upgrade all of them things can become more unstable :) But I am fine with updates as long as tests pass.

**Comment by prathamesh-sonpatki** [comment]
@JuanitoFatas I fixed the failing tests here - https://github.com/codetriage/codetriage/commit/823df04175fa733f90203c1805d669f12cff7460. The issue was in warden gem :)

**Comment by prathamesh-sonpatki** [comment]
Closing in favor of https://github.com/codetriage/codetriage/pull/444 which has this commit. Thanks @jacarandang :sparkles:

**Pull request closed without merging**